samples/hello-world/dotnetcore2.0/build.cake

```diff
@@ -0,0 +1,231 @@
+        foreach(var project in projects)
+        {
+			var projectName =  project
+				.Split(new [] {'/'}, StringSplitOptions.RemoveEmptyEntries)
+				.Last()
+				.Replace(".csproj", string.Empty);
+
+            var outputDirectory = System.IO.Path.Combine(publishDir, projectName);
+
+            var msBuildSettings = new DotNetCoreMSBuildSettings 
+            {
+                TreatAllWarningsAs = MSBuildTreatAllWarningsAs.Error,
+                Verbosity = dotNetCoreVerbosity
+            };
+
+			var settings = new DotNetCorePublishSettings
+			{
+				Configuration = configuration,
+				MSBuildSettings = msBuildSettings,
+				NoRestore = true,
+				OutputDirectory = outputDirectory,
+				Verbosity = dotNetCoreVerbosity
+			};
+
+            Information("Publishing '{0}'...", projectName);
+            DotNetCorePublish(project, settings); 
+            Information("'{0}' has been published.", projectName);
+        }
+    });
+
+Task("Pack")
+	.Description("Packs all the different parts of the project.")
+	.Does(() => 
+    {
+		foreach(var project in projects)
+        {
+			var projectName = GetProjectName(project);
+
+			Information("Packing '{0}'...", projectName);
+			var path = System.IO.Path.Combine(publishDir, projectName);
+			var files = GetFiles(path + "/*.*");
+			Zip(
+				path, 
+				System.IO.Path.Combine(artifactsDir, $"{projectName}.zip"),
+				files);
+			Information("'{0}' has been packed.", projectName);
+        }
+    });
+
+Task("Run-Local")
+	.Description("Runs all the acceptance tests locally.")
+	.Does(() => 
+    {
+        var settings = new ProcessSettings
+        { 
+            Arguments = "local invoke \"HelloWorld\" -e event.json",
+        };
+        
+        Information("Starting the SAM local...");
+        using(var process = StartAndReturnProcess("sam", settings))
+        {
+            process.WaitForExit();
+            Information("Exit code: {0}", process.GetExitCode());
+        }
+        Information("SAM local has finished.");
+    });
+
+///////////////////////////////////////////////////////////////////////////////
+// TARGETS
+///////////////////////////////////////////////////////////////////////////////
+
+Task("Package")
+    .Description("This is the task which will run if target Package is passed in.")
+    .IsDependentOn("Clean")
+    .IsDependentOn("Restore")
+    .IsDependentOn("Build")
+    .IsDependentOn("Publish")
+    .IsDependentOn("Pack")
+    .Does(() => { Information("Package target ran."); });
+
+Task("Run")
+    .Description("This is the task which will run if target Run is passed in.")
+    .IsDependentOn("Clean")
+    .IsDependentOn("Restore")
+    .IsDependentOn("Build")
+    .IsDependentOn("Publish")
+    .IsDependentOn("Pack")
+    .IsDependentOn("Run-Local")
+    .Does(() => { Information("Run target ran."); });
+    
+Task("Default")
+    .Description("This is the default task which will run if no specific target is passed in.")
+    .IsDependentOn("Default");
```

Comment:
This causes a `Reflexive edges in graph are not allowed` exception (the task being dependant on itself).